sources/param_get_envvars.c

```diff
@@ -3,38 +3,130 @@
 /*                                                        :::      ::::::::   */
 /*   param_get_envvars.c                                :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/10/06 20:45:38 by inikulin          #+#    #+#             */
-/*   Updated: 2024/11/15 14:53:09 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/12/24 15:52:08 by taretiuk         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "minishell.h"
-#ifdef MOCK_TANIA
 
-/* returns error code */
-int	param_get_envvars(t_param *param)
+static int	calc_len(const char *s, int *start, int *len_1, int *len_2)
 {
-	if (!ft_dlist_add_back_s(&param->envvar_path_head, "/usr/local/sbin")
-		|| !ft_dlist_add_back_s(&param->envvar_path_head, "/usr/local/bin")
-		|| !ft_dlist_add_back_s(&param->envvar_path_head, "/usr/sbin")
-		|| !ft_dlist_add_back_s(&param->envvar_path_head, "/usr/bin")
-		|| !ft_dlist_add_back_s(&param->envvar_path_head, "/sbin/bin")
-		|| ft_mapss_add(param->envvars, "HOME", "/home/ioann")
-		|| ft_mapss_add(param->envvars, "some", "BODYONCETOLDME")
-	)
+	int	i;
+	int	delim;
+
+	i = 0;
+	*len_1 = 0;
+	*len_2 = 0;
+	delim = 0;
+	while (s[i])
+	{
+		if (s[i] == '=' && delim == 0)
+		{
+			delim = 1;
+			*len_1 = i;
+			*start = i + 1;
+		}
+		i++;
+	}
+	if (delim)
+		*len_2 = i - *len_1 - 1;
+	else
+		*len_2 = 0;
+	return (0);
+}
+
+char	**split_env(char *s)
+{
+	int		key_start;
+	int		len_1;
+	int		len_2;
+	char	**res;
+
+	calc_len(s, &key_start, &len_1, &len_2);
+	res = (char **)ft_calloc_if(sizeof(char *) * 3, 1);
```

Comment:
if it's same size every time, we might as well use char *res[2] without allocating memory dynamically